### ISSUE TITLE
[master] cmake-tools.nvim/lua/cmake-tools/scratch.lua:8: Failed to rename buffer

### ISSUE THREAD

**Issue by hotung1027** [Civitasv/cmake-tools.nvim#199]
**Bug description**
On Vim Enter,
Lsp attach and trigger cmake-tools to setup
And then error pops out

CMake-Tools cannot function properly afterwards.

**Steps to reproduce**

<details><summary>Minimal configuration</summary>
````
'stevearc/overseer.nvim'
'Civitasv/cmake-tools.nvim'
"hrsh7th/nvim-cmp"
````
</details>

**Expected behavior**

**Screenshots**
![Screenshot 2024-02-24 at 12 02 51 AM](https://github.com/Civitasv/cmake-tools.nvim/assets/20833241/287ff398-b3b6-4741-95f1-f6999159ce08)
![Screenshot 2024-02-24 at 12 27 24 AM](https://github.com/Civitasv/cmake-tools.nvim/assets/20833241/e7a25792-6384-42a1-8d72-3355a949ecd2)

**Environment**

- OS: Macos Sonoma
- Desktop environment: MacOS apple Silicon
- Neovim version:
````
❯ nvim --version
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1703358377

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.9.5/share/nvim"
````
- Plugins commit hash: 055d7bb37d5c4038ce1e400656b6504602934ce7 

**Contents in the \*cmake-tools\* buffer**

Cannot populate since the error produce from setup
Trying to checkout to older version and see the problem fix from which hash

**Comment by hotung1027** [comment]
Update after manual checkout to cd5db8b22de4, everything works smoothly.

Maybe I miss some setup after the breaking changes?

**Comment by Civitasv** [comment]
Do you call `setup` for cmake-tools.nvim? And can you check if the buffer `*cmake-tools*` exists?

**Comment by hotung1027** [comment]
> Do you call `setup` for cmake-tools.nvim? And can you check if the buffer `*cmake-tools*` exists?
Yep, I invoke cmake-tools.nvim to setup when file type is `c`,`cpp`.
I did try setup at Lazy.nvim load, both produce the same result.
And I checked the buffer *cmake-tools* not exists, through vim bufferlist.

**Comment by hotung1027** [comment]
I've checked that after I checkout to a version in which scratch.lua exists, it works smoothly, should be somewhere before 8e4e7f28652b53d90324703765a0827a95caa4e2.

I am not sure I hitting this behaviour because of my neovim version or what.

**Comment by Civitasv** [comment]
It should work now, use the latest commit.
